plugins/microsoft_log_analytics/icon_microsoft_log_analytics/util/tools.py

```diff
@@ -91,3 +93,67 @@ def return_non_empty_query_output(query_response_input: dict) -> dict:
     """
 
     return {"tables": list(filter(lambda x: x.get("rows"), query_response_input.get("tables")))}
+
+
+def remove_keys_from_saved_search(
+    sub_dictionaries: List[str], keys: List[str], input_list_of_dicts_or_dict: Union[List[dict], dict]
+) -> Union[List[dict], dict]:
+    """Method allows to remove version key from List All Saved Searches action's output.
+
+    :param input_list_of_dicts_or_dict: Input dictionary from which all keys will be removed
+    :type input_list_of_dicts_or_dict: List[dict]
+
+    :returns: API call function data without keys in dictionaries and sub-dictionaries
+    :rtype: List[dict]
+    """
+
+    if isinstance(input_list_of_dicts_or_dict, list):
+        for key in keys:
+            for dictionary in sub_dictionaries:
+                for element in input_list_of_dicts_or_dict:
+                    if key in element:
+                        element.pop(key)
+                    if key in element.get(dictionary):
+                        element.get(dictionary).pop(key)
+    elif isinstance(input_list_of_dicts_or_dict, dict):
+        for key in keys:
+            for dictionary in sub_dictionaries:
+                if key in input_list_of_dicts_or_dict:
+                    input_list_of_dicts_or_dict.pop(key)
+                if key in input_list_of_dicts_or_dict.get(dictionary):
+                    input_list_of_dicts_or_dict.get(dictionary).pop(key)
+    return input_list_of_dicts_or_dict
+
+
+def add_names_to_saved_searches_list(input_saved_search_list: List[dict]) -> List[dict]:
+    """Add names to all elements in list of Saved Searches if doesn't exist.
+
+    :param input_saved_search_list: Input list of dictionary that contains Saved Searches
+    :type input_saved_search_list: List[dict]
+
+    :returns: API call function data with added name keys if neccessary
+    :rtype: List[dict]
+    """
+
+    for element in input_saved_search_list:
+        if "name" not in element:
+            element["name"] = element.get("id").split("/")[-1]
+    return input_saved_search_list
+
+
+def return_non_empty(input_dict: Dict[str, Any]) -> Dict[Any, Any]:
```

Comment:
Is it desired that this method can only accept an input_dict where the keys are strings? Can we not handle dicts where the keys are ints?

Reply:
in that case every single JSON key is `str`